packages/relayer/processor/process_message.go

```diff
@@ -461,3 +452,55 @@ func (p *Processor) saveMessageStatusChangedEvent(
 
 	return nil
 }
+
+func (p *Processor) setGasTipOrPrice(ctx context.Context, auth *bind.TransactOpts) error {
+	gasTipCap, err := p.destEthClient.SuggestGasTipCap(ctx)
+	if err != nil {
+		if IsMaxPriorityFeePerGasNotFoundError(err) {
+			auth.GasTipCap = FallbackGasTipCap
+		} else {
+			gasPrice, err := p.destEthClient.SuggestGasPrice(context.Background())
+			if err != nil {
+				return errors.Wrap(err, "p.destBridge.SuggestGasPrice")
+			}
+			auth.GasPrice = gasPrice
+		}
+	} else {
+		auth.GasTipCap = gasTipCap
+
+		return nil
+	}
+
+	return nil
+}
+
+func (p *Processor) getCost(ctx context.Context, auth *bind.TransactOpts) (*big.Int, error) {
+	if auth.GasTipCap != nil {
+		blk, err := p.destEthClient.BlockByNumber(ctx, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		cfg := params.NetworkIDToChainConfigOrDefault(p.destChainId)
```

Comment:
we only need to use this in the `else` clause below, its not used in the `if p.taikoL2 != nil` clause, so we can move it there, above `baseFee = eip1559.CalcBaseFee`.